docs/support_files/world.md

```diff
@@ -36,32 +51,85 @@ $ cucumber-js --world-parameters '{"appUrl":"http://localhost:3000/"}'
 
 This option is repeatable, so you can use it multiple times and the objects will be merged with the later ones taking precedence.
 
+Using JSON literals on the command line is tedious, but you can use [profiles](profiles.md) to create aliases for them.
+
 This data is then available on `this.parameters` from your hooks and steps.
 
-## Custom World
+## Custom Worlds
+
+You might also want to have methods on your World that hooks and steps can access to keep their own code simple. To do this, you can provide your own World class with its own properties and methods that help with your instrumentation, and then call `setWorldConstructor` to tell Cucumber about it.
+
+Let's walk through a typical scenario, setting up world that manages a browser context. We'll use the ES6 module syntax for this example.  First, let's set up our custom world. Class files should not be loaded as steps - they should be imported. So in this example we'll presume it is in a classes folder next to the steps folder.
+
+###### CustomWorld.js
+```javascript
+import { World } from '@cucumber/cucumber';
+import seleniumWebdriver from "selenium-webdriver";
+
+export default class extends World {
+  driver = null;
+
+  /*
+   * A constructor is only needed if you have custom actions
+   * to take after the Cucumber parses the options.
+   */
+  constructor(options) {
+    super(options);
+    // Custom actions go here.
+  }
+
+  /*
+   * Constructors cannot be asynchronous! To work around this we'll
+   * use an init method with the Before hook
+   */
+  async init(scenario) {
```

Comment:
Good improvement 👍

Reply:
Thanks. Added a bit more, is it too much or good. I think the above at this point is all I can think of.

Reply:
Thanks.